test/jdk/java/lang/reflect/code/bytecode/TestBytecode.java

```diff
@@ -275,6 +279,67 @@ static int whileLoop(int i, int n) { int
         return counter;
     }
 
+    public interface Func {
+        int apply(int a);
+    }
+
+    public interface QuotableFunc extends Quotable {
+        int apply(int a);
+    }
+
+    static int consume(int i, Func f) {
+        return f.apply(i + 1);
+    }
+
+    static int consumeQuotable(int i, QuotableFunc f) {
+        return f.apply(i + 1);
```

Comment:
For now perhaps assert that the result of `f.quoted()` and `f.quoted().op()` are non-null and the latter is an instance of `LambdaOp`. A separate test is needed to assert on the code model and captured parameters, we could add that in another PR.

Reply:
I've added null and instanceof checks.